### PR TITLE
Fix: Propagate context with DownloadContents

### PR DESCRIPTION
**Pull request by HakShak**
Main motivation is tracing as this can be the most time consuming operation.

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @HakShak !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by tomfeigin — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @tomfeigin !
Merging.

**Pull request merged**
